api/v1alpha1/nodepool_types.go

```diff
@@ -293,6 +293,12 @@ type NodePoolPlatform struct {
 	// +optional
 	// +immutable
 	Kubevirt *KubevirtNodePoolPlatform `json:"kubevirt,omitempty"`
+
+	// Agent specifies the configuration used when using Agent platform.
+	//
+	// +optional
+	// +immutable
+	Agent *AgentNodePoolPlatform `json:"agent,omitempty"`
```

Comment:
Actually we should drop the immutable for the other platforms. A change into the platform spec tiggers a rolling upgrade.

Reply:
Removed immutable for all platforms